Mathlib/Analysis/NormedSpace/Unitization.lean

```diff
@@ -0,0 +1,248 @@
+-- ummmm... what? why does Lean need me to fill in these instances?
+
+attribute [local instance] Unitization.normedRingAux
+
+/-- Pull back the normed algebra structure from `𝕜 × (A →L[𝕜] A)` to `Unitization 𝕜 A` using the
+algebra homomorphism `Unitization.splitMul 𝕜 A`. This uses the wrong `NormedRing` instance (i.e.,
+`Unitization.normedRingAux`), so we only use it as a local instance to build the real one.-/
+@[reducible]
+noncomputable def normedAlgebraAux : NormedAlgebra 𝕜 (Unitization 𝕜 A) :=
+  NormedAlgebra.induced 𝕜 (Unitization 𝕜 A) (𝕜 × (A →L[𝕜] A)) (splitMul 𝕜 A)
+
+attribute [local instance] Unitization.normedAlgebraAux
+
+theorem norm_def (x : Unitization 𝕜 A) : ‖x‖ = ‖splitMul 𝕜 A x‖ :=
+  rfl
+
+/-- This is often the more useful lemma to rewrite the norm as opposed to `Unitization.norm_def`. -/
+theorem norm_eq_sup (x : Unitization 𝕜 A) :
+    ‖x‖ = ‖x.fst‖ ⊔ ‖algebraMap 𝕜 (A →L[𝕜] A) x.fst + mul 𝕜 A x.snd‖ := by
+  rw [norm_def, splitMul_apply, Prod.norm_def, sup_eq_max]
+
+variable (𝕜 A)
+
+/-- The identity map between `Unitization 𝕜 A` and `𝕜 × A` as an `AddEquiv`. -/
+def addEquiv : Unitization 𝕜 A ≃+ 𝕜 × A :=
+  AddEquiv.refl _
+
+variable {𝕜 A}
+
+theorem lipschitzWith_addEquiv :
+    LipschitzWith 2 (Unitization.addEquiv 𝕜 A) := by
+  rw [← Real.toNNReal_ofNat]
+  refine AddMonoidHomClass.lipschitz_of_bound (Unitization.addEquiv 𝕜 A) 2 fun x => ?_
+  rw [norm_eq_sup, Prod.norm_def]
+  refine' max_le ?_ ?_
+  · rw [sup_eq_max, mul_max_of_nonneg _ _ (zero_le_two : (0 : ℝ) ≤ 2)]
+    exact le_max_of_le_left ((le_add_of_nonneg_left (norm_nonneg _)).trans_eq (two_mul _).symm)
+  · nontriviality A
+    rw [two_mul]
+    calc
+      ‖x.snd‖ = ‖mul 𝕜 A x.snd‖ :=
+        .symm <| (isometry_mul 𝕜 A).norm_map_of_map_zero (map_zero _) _
+      _ ≤ ‖algebraMap 𝕜 _ x.fst + mul 𝕜 A x.snd‖ + ‖x.fst‖ := by
+        simpa only [add_comm _ (mul 𝕜 A x.snd), norm_algebraMap'] using
+          norm_le_add_norm_add (mul 𝕜 A x.snd) (algebraMap 𝕜 _ x.fst)
+      _ ≤ _ := add_le_add le_sup_right le_sup_left
+
+theorem antilipschitzWith_addEquiv :
+    AntilipschitzWith 2 (addEquiv 𝕜 A) := by
+  refine AddMonoidHomClass.antilipschitz_of_bound (addEquiv 𝕜 A) fun x => ?_
+  rw [norm_eq_sup, Prod.norm_def, NNReal.coe_two]
+  refine max_le ?_ ?_
+  · rw [mul_max_of_nonneg _ _ (zero_le_two : (0 : ℝ) ≤ 2)]
+    exact le_max_of_le_left ((le_add_of_nonneg_left (norm_nonneg _)).trans_eq (two_mul _).symm)
+  · nontriviality A
+    calc
+      ‖algebraMap 𝕜 _ x.fst + mul 𝕜 A x.snd‖ ≤ ‖algebraMap 𝕜 _ x.fst‖ + ‖mul 𝕜 A x.snd‖ :=
+        norm_add_le _ _
+      _ = ‖x.fst‖ + ‖x.snd‖ := by
+        rw [norm_algebraMap', (AddMonoidHomClass.isometry_iff_norm (mul 𝕜 A)).mp (isometry_mul 𝕜 A)]
+      _ ≤ _ := (add_le_add (le_max_left _ _) (le_max_right _ _)).trans_eq (two_mul _).symm
+
+open Bornology Filter
+open scoped Uniformity
+
+theorem uniformity_eq_aux :
+    @uniformity _ (instUniformSpaceProd.comap <| addEquiv 𝕜 A) = 𝓤 (Unitization 𝕜 A) := by
```

Comment:
I think you could use [Topology.«term𝓤[_]»](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/UniformSpace/Basic.html#Topology.%C2%ABterm%F0%9D%93%A4[_]%C2%BB) to make this line a bit more readable

Reply:
I thought that existed but it didn't work because I didn't have that scope open. Grrr... Why is it in the Topology namespace?

Reply:
It shouldn't, unless we decide to put all uniformity notations in `Topology`, that would make sense too.

Reply:
#6565